packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -321,51 +320,25 @@ export function useConnections({
         );
       }
     },
-    [onConnected, saveConnectionInfo, persistOIDCTokens]
+    [onConnected, persistOIDCTokens, saveConnectionInfo]
   );
 
   useEffect(() => {
-    void connectWithAutoConnectInfoIfAvailable();
-    async function connectWithAutoConnectInfoIfAvailable() {
-      let autoConnectInfo: ConnectionInfo | undefined;
-      try {
-        autoConnectInfo = await connectionStorage.getAutoConnectInfo?.();
-        if (autoConnectInfo) {
-          log.info(
-            mongoLogId(1_001_000_160),
-            'Connection Store',
-            'Performing automatic connection attempt'
-          );
-          dispatch({
-            type: 'set-active-connection',
-            connectionInfo: autoConnectInfo,
-          });
-          void connect(autoConnectInfo, false);
-        }
-      } catch (error) {
-        onConnectionFailed(autoConnectInfo ?? null, error as Error);
-        log.error(
-          mongoLogId(1_001_000_290),
-          'Connection Store',
-          'Error performing connection attempt using auto connect info',
-          {
-            error: (error as Error).message,
-          }
-        );
-
-        dispatch({
-          type: 'connection-attempt-errored',
-          connectionErrorMessage: (error as Error).message,
-        });
-      }
+    if (connectionStorage.getAutoConnectInfo) {
+      void connect(
+        connectionStorage.getAutoConnectInfo.bind(connectionStorage),
+        false
+      ).catch(() => {
+        // noop, we're already logging in the connect method
```

Comment:
I've seen this twice when we call the connect method. If we are not going to handle the exception anywhere, should we just let connect swallow it and do the handling there?

Reply:
Sidebar still having parts its own connection flow right now handles the exception in it's own way so we allow connect method to throw and ignore the error where its irrelevant. After we move what's left of the connection logic from the sidebar to the compass-connections, we should be able to remove it and indeed not throw inside connect when errors happen